streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -351,7 +351,8 @@ private void processEarly(final K key, final V value, final long inputRecordTime
             }
 
             if (combinedWindow == null) {
-                final TimeWindow window = new TimeWindow(0, windows.timeDifferenceMs());
+                // created a [start, end] time interval window via SessionWindow
+                final SessionWindow window = new SessionWindow(0, windows.timeDifferenceMs());
```

Comment:
add a comment here to avoid confusion.

Reply:
I would prefer to _first_ rename existing windows and not merge this PR using `SessionWindows` within `SlidingWindowAggregate`...